### PR TITLE
CSettings: change settings definitions to constexpr

### PR DESCRIPTION
**Pull request by lrusak**
This fixes a segfault that I seem to have on LibreELEC. I'm not sure why I cannot reproduce it on my workstation.

```
#0  0x00007f347e3176db in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::string const&) () from /usr/lib/libstdc++.so.6
#1  0x00000000009df12f in std::pair<std::string const, std::string>::pair<char const (&) [8], true> (__y=..., __x=..., this=0x7fff4e1c64a0)
    at /home/lukas/git/libreelec/build.LibreELEC-Generic.x86_64-9.80-devel/toolchain/x86_64-libreelec-linux-gnu/include/c++/10.2.0/ext/new_allocator.h:89
#2  __static_initialization_and_destruction_0 (__priority=65535, __initialize_p=1) at ../xbmc/video/VideoThumbLoader.cpp:259
#3  0x0000000001acfef5 in __libc_csu_init (argc=argc@entry=1, argv=argv@entry=0x7fff4e1c66e8, envp=0x7fff4e1c66f8) at elf-init.c:88
#4  0x00007f347e422c4a in __libc_start_main (main=0x9d1540 <main(int, char**)>, argc=1, argv=0x7fff4e1c66e8, init=0x1acfeb0 <__libc_csu_init>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fff4e1c66d8) at ../csu/libc-start.c:264
#5  0x0000000000a294ca in _start () at ../sysdeps/x86_64/start.S:120
```

Basically the program segfaults before `main()` is even called due to a static initializer failing.

This code block here https://github.com/xbmc/xbmc/blob/master/xbmc/video/VideoThumbLoader.cpp#L259-L266 is the problem and the `CSettings` definitions may not be initialized before this anonymous namespace in `VideoThumbLoader.cpp`.

I was able to fix it by simply moving the map into the `CVideoThumbLoader` class as a static member but this didn't seem right.

I ended up with this fix in this PR so that they are available from compile time.

I've also added a `todo` for c++17 as the defining of the constexpr variables in the cpp file is not so nice. See here for more info: https://stackoverflow.com/a/28846608

### CODE REVIEW

File: xbmc/settings/Settings.cpp [hunk 1]
```diff
@@ -60,388 +60,379 @@
...
-const std::string CSettings::SETTING_ADDONS_ALLOW_UNKNOWN_SOURCES = "addons.unknownsources";
-const std::string CSettings::SETTING_ADDONS_UPDATEMODE = "addons.updatemode";
-const std::string CSettings::SETTING_ADDONS_MANAGE_DEPENDENCIES = "addons.managedependencies";
-const std::string CSettings::SETTING_GENERAL_ADDONFOREIGNFILTER = "general.addonforeignfilter";
-const std::string CSettings::SETTING_GENERAL_ADDONBROKENFILTER = "general.addonbrokenfilter";
-const std::string CSettings::SETTING_SOURCE_VIDEOS = "source.videos";
-const std::string CSettings::SETTING_SOURCE_MUSIC = "source.music";
-const std::string CSettings::SETTING_SOURCE_PICTURES = "source.pictures";
+//! @todo: remove in c++17
+constexpr const char* CSettings::SETTING_LOOKANDFEEL_SKIN;
```
Comment (garbear) [inline xbmc/settings/Settings.cpp:64]:
```suggestion
//! @todo: remove in c++17
///@{
constexpr const char* CSettings::SETTING_LOOKANDFEEL_SKIN;
```

File: xbmc/settings/Settings.cpp [hunk 2]
```diff
@@ -60,388 +60,379 @@
...
+constexpr const char* CSettings::SETTING_ADDONS_SHOW_RUNNING;
+constexpr const char* CSettings::SETTING_ADDONS_ALLOW_UNKNOWN_SOURCES;
+constexpr const char* CSettings::SETTING_ADDONS_UPDATEMODE;
+constexpr const char* CSettings::SETTING_ADDONS_MANAGE_DEPENDENCIES;
+constexpr const char* CSettings::SETTING_GENERAL_ADDONFOREIGNFILTER;
+constexpr const char* CSettings::SETTING_GENERAL_ADDONBROKENFILTER;
+constexpr const char* CSettings::SETTING_SOURCE_VIDEOS;
+constexpr const char* CSettings::SETTING_SOURCE_MUSIC;
+constexpr const char* CSettings::SETTING_SOURCE_PICTURES;
+//! @todo: remove in c++17
```
Comment (garbear) [inline xbmc/settings/Settings.cpp:435]:
```suggestion
///@}
```

**Review by garbear — APPROVED** [review]
@AlwinEsch am I right about the doxy format suggestion? @lrusak It doesn't matter for this PR, Just asking if I'm doing it right.

**Comment by AlwinEsch** [comment]
> @AlwinEsch am I right about the doxy format suggestion? @lrusak It doesn't matter for this PR, Just asking if I'm doing it right.

Partly yes, the `@{ \n \n @}` is mainly about grouping (https://www.doxygen.nl/manual/grouping.html) and where need a part to define, not tested now but maybe the `\name` is enough.

**Review by yol — APPROVED** [review]

**Comment by lrusak** [comment]
merge?

**Pull request merged**